### ISSUE TITLE
Parquet-avro fails to decode array of record with a single field name "element" correctly

### ISSUE THREAD

**Issue by asfimport** [apache/parquet-java#1976]
Found this issue while investigating SPARK-16344.

For the following Parquet schema

```
message root {
  optional group f (LIST) {
    repeated group list {
      optional group element {
        optional int64 element;
      }
    }
  }
}
```

parquet-avro decodes it as something like this:

```
record SingleElement {
  int element;
}

record NestedSingleElement {
  SingleElement element;
}

record Spark16344Wrong {
  array<NestedSingleElement> f;
}
```

while correct interpretation should be:

```
record SingleElement {
  int element;
}

record Spark16344 {
  array<SingleElement> f;
}
```

The reason is that the `element` syntactic group for LIST in

```
<list-repetition> group <name> (LIST) {
  repeated group list {
    <element-repetition> <element-type> element;
  }
}
```

is recognized as a record field named `element`. The problematic code lies in [`AvroRecordConverter.isElementType()`](https://github.com/apache/parquet-mr/blob/bd0b5af025fab9cad8f94260138741c252f45fc8/parquet-avro/src/main/java/org/apache/parquet/avro/AvroRecordConverter.java#L858). We should probably check the standard 3-level layout first before falling back to the legacy 2-level layout.

**Reporter**: [Cheng Lian](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=lian+cheng) / @liancheng
**Assignee**: [Ryan Blue](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rdblue) / @rdblue
#### Related issues:
- [Avro's isElementType() change breaks the reading of some parquet(1.8.1) files](https://issues.apache.org/jira/browse/PARQUET-1681) (is caused by)
- [Array of struct with a single field name "element" can't be decoded from Parquet files written by Spark 1.6+](https://issues.apache.org/jira/browse/SPARK-16344) (is related to)
#### PRs and other links:
- [PR #352](https://github.com/apache/parquet-mr/pull/352)

<sub>**Note**: *This issue was originally created as [PARQUET-651](https://issues.apache.org/jira/browse/PARQUET-651). Please see the [migration documentation](https://issues.apache.org/jira/browse/PARQUET-2502) for further details.*</sub>

**Comment by asfimport** [comment]
[Ryan Blue](https://issues.apache.org/jira/browse/PARQUET-651?#comment-15371282) / @rdblue:
Fixed in PR #352 by converting the file's element schema to Avro and comparing it with the expected schema. Thanks to [~lian cheng], who came up with the solution for SPARK-16344!

**Comment by asfimport** [comment]
[Ryan Blue](https://issues.apache.org/jira/browse/PARQUET-651?#comment-15381545) / @rdblue:
Merged #352. Thanks for reviewing, [~lian cheng]!
